fair_mango/dataset/dataset.py

```diff
@@ -118,7 +135,8 @@ def __init__(
             )
         self.n_groups: int = len(self.groups)
         self.groups_data: list[dict] = []
-        self.groups_predicted_target: list[dict] = []
+        self.groups_real_target: list[dict] | None = None
+        self.groups_predicted_target: list[dict] | None = None
```

Comment:
can you specify the type of `dict` ?

Reply:
I try to avoid detailing typing when it can hold values of different data types. For instance `self.groups_real_target` and `self.groups_predicted_target` can hold values of type `list[dict[str, np.ndarray | pd.Series | pd.DataFrame]]`. I added it in the code too.